### ISSUE TITLE
istioctl experimental describe lacks gateway infomation when k8s gateway api is used

### ISSUE THREAD

**Issue by hanxiaop** [istio/istio#46160]
It will display:
```
istioctl x describe svc httpbin
Service: httpbin
   Port: http 8000/HTTP targets pod port 80
Skipping Gateway information (no ingress gateway pods)
```

I think this can be handled similar to ingress pods.

**Comment by hanxiaop** [comment]
Found it also has issues when using injected gateway.

**Comment by hanxiaop** [comment]
not stale

**Comment by hanxiaop** [comment]
@nicole-lihui would you like to work on this?

**Comment by nicole-lihui** [comment]
Sure, I'd be happy to work on this!

**Comment by nicole-lihui** [comment]
> Found it also has issues when using injected gateway.

Does `injected gateway ` refer to a custom deployed or configured gateway(similar to helm custom installation)?

**Comment by hanxiaop** [comment]
> > Found it also has issues when using injected gateway.
> 
> Does `injected gateway ` refer to a custom deployed or configured gateway(similar to helm custom installation)?

Yes, rather than a fixed image, it is injected by the gateway template. See https://istio.io/latest/docs/setup/additional-setup/gateway/ as an example.

**Comment by nicole-lihui** [comment]
> @nicole-lihui Yes, rather than a fixed image, it is injected by the gateway template. See https://istio.io/latest/docs/setup/additional-setup/gateway/ as an example.

great, and I also noticed other issues heres. ref:  https://github.com/istio/istio/pull/49856
